PriceKataCalc/Program.cs

```diff
@@ -89,19 +104,55 @@ private double UPCDiscount()
             else return 0;
         }
 
-        internal object calcAllDiscount()
+        public double calcAllDiscount()
         {
             return this.calcDiscount() + this.calcUPCDiscount();   
         }
+
+        public double calcCosts()
+        {
+            double sum = 0;
+            foreach(cost i in this.Costs)
+            {
+                if (i.isPercentage) sum += this.price;
+                else sum += i.FlatCost;
+            }
+            return sum;
+        }
     }
      class Discount
     {
         public bool beforeTax { set; get; }
+        public static bool  CombiningAddictive { get; set; }
         public double discountValue { set; get; }
-        public Discount(bool beforeTax, double discountValue)
+        public Discount(bool beforeTax,bool Combining, double discountValue)
         {
             this.beforeTax = beforeTax;
+            CombiningAddictive = Combining;
             this.discountValue = discountValue;
         }
-    }
+        }
+        class cost
+        {
+            public bool isPercentage;
```

Comment:
Is there another way to represent this?

Reply:
maybe using an interface for the cost will solve this issue